src/inmanta/module.py

```diff
@@ -2958,20 +2958,17 @@ def get_plugin_dir(self) -> Optional[str]:
         raise NotImplementedError()
 
     def _list_python_files(self, plugin_dir: str) -> list[str]:
-        """
-        Generate a list of all Python files in the given plugin directory.
-        This method prioritizes .pyc files over .py files, uses caching to avoid duplicate directory walks,
-        and only considers directories that are Python packages.
-        """
+        """Generate a list of all python files, including namespace packages and excluding the model directory."""
         # Return cached results if this directory has been processed before
+        model_dir_path: str = os.path.join(plugin_dir, "inmanta_plugins", self.name, "model")
         if plugin_dir in self._dir_cache:
             return self._dir_cache[plugin_dir]
 
         files: dict[str, str] = {}
 
         for dirpath, dirnames, filenames in os.walk(plugin_dir):
```

Comment:
I don't think we can afford to use `os.walk` here, as it doesn't allow us to stop scanning below the model dir. 

(there can be >10k files below it). 

I would look at the implementation of os.walk, steal the relevant parts and make is so that you can simply stop going down below the model folder.